activejob/lib/active_job/logging.rb

```diff
@@ -87,12 +87,25 @@ def queue_name(event)
         def args_info(job)
           if job.arguments.any?
             ' with arguments: ' +
-              job.arguments.map { |arg| arg.try(:to_global_id).try(:to_s) || arg.inspect }.join(', ')
+              job.arguments.map { |arg| serialize_arg(arg).inspect }.join(', ')
           else
             ''
           end
         end
 
+        def serialize_arg(arg)
+          case arg
+          when Hash
+            arg.each_with_object({}) { |(key, value), hash| hash[key] = serialize_arg(value)}
+          when Array
+            arg.map { |value| serialize_arg(value) }
+          when GlobalID::Identification
+            arg.to_global_id.to_s
```

Comment:
As you can see in `Arguments` `to_global_id` can raise for an unpersisted model, so let's keep the `try` in there. We can even use a block:

```ruby
arg.try { to_global_id.to_s }
```